internal/docstore/firedocstore/query.go

```diff
@@ -114,7 +224,28 @@ func (c *collection) queryToProto(q *driver.Query) (*pb.StructuredQuery, error)
 	}
 	// TODO(jba): order
 	// TODO(jba): cursors (start/end)
-	return p, nil
+	return p, localFilters, nil
+}
+
+// splitFilters separates the list of query filters into those we can send to the Firestore service,
+// and those we must evaluate here on the client.
+func splitFilters(fs []driver.Filter) (sendToFirestore, evaluateLocally []driver.Filter) {
+	// Enforce that only one field can have an inequality.
+	var rangeFP []string
+	for _, f := range fs {
+		if f.Op == driver.EqualOp {
+			sendToFirestore = append(sendToFirestore, f)
+		} else {
+			if rangeFP == nil || fpEqual(rangeFP, f.FieldPath) {
+				// Multiple inequality filters on the same field are OK.
```

Comment:
The docstring at the top of the package states only the first where clause with inequality is passed. Should that one be changed?

Reply:
Thanks, I clarified.